openamp/overview.rst

```diff
@@ -6,69 +6,223 @@ Project Overview
+********************
+
+The OpenAMP framework utilizes a number of distinct components to achieve the :ref:`OpenAMP Fundamentals<openamp-fundamentals-work-label>`, with most of them using or derived from existing standards or frameworks.
+
+The components comprising OpenAMP are:
+
+.. csv-table::
+   :header: "AMP Component", "AMP Fundamentals"
+   :widths: 50, 60
+
+    :ref:`Remoteproc<overview-remoteproc-work-label>`, Resource Assignment and Runtime Control
+    Resource Table, Resource Assignment
+    :ref:`RPMsg<overview-rpmsg-work-label>`, IPC
+    :ref:`Virtio<overview-rpmsg-work-label>`, Hypervisor and Hardware Abstraction
+    :ref:`Proxy<overview-proxy-rpc-work-label>`, IPC for File Input Output (IO)
+    :ref:`Libmetal<overview-proxy-libmetal-label>`, Hardware Abstraction
 
-To accomplish the above, OpenAMP is divided into the following efforts:
+
+The :ref:`topology<topology-work-label>` is limited to host controller to remote system but otherwise open to the implementation.
+
+The architecture is exemplified below via a daisy chained topology, with the center processor being both remote and host controller for the next processor in the chain. This is an alternate topology to the previous example in the :ref:`OpenAMP Fundamentals<openamp-fundamentals-work-label>` section.
+
+..  image::  ../images/architecture/overview-architecture.svg
+
+In an attempt to keep the diagrams clear, not all OpenAMP components are drawn on each processor.
+
+.. _overview-remoteproc-work-label:
+
+RemoteProc
+==========
+
+:ref:`Runtime Control<runtime-control-work-label>` is part of the :ref:`Life Cycle Management (LCM)<lcm-work-label>` of the remote(s) and performed through the :ref:`remoteproc<lcm-work-label>` component which allows for the loading of the firmwares to the remote processor and starting and stopping the remote.
+
+:ref:`Resource Assignment<resource-assignment-work-label>` is also achieved through the :ref:`remoteproc<lcm-work-label>` component utilizing a Resource Table, which provides the memory and peripheral allocation as well as information for establishing the IPC between associated processors.
+
+..  image::  ../images/architecture/overview-architecture-remoteproc.svg
+
+.. _overview-rpmsg-work-label:
+
+RPMsg and Virtio
+================
+
+Standardization of the IPC is promoted by the OpenAMP project through the use of :ref:`RPMsg <rpmsg-protocol-work-label>`, using `Open Standard Virtio Devices <https://docs.oasis-open.org/virtio/virtio/v1.3/virtio-v1.3.html>`_ as a HW abstraction or MAC layer.
+
+This abstraction, using virtio, means that the implementer can optionally use :ref:`resource isolation<resource-isolation-work-label>` (e.g. using a hypervisor or secure context), which is exemplified by the first processor in the architecture diagram. The other two remotes are in what is referred to as a hypervisorless-virtio setup because they are using virtio (virtual io) as an abstraction layer but without a hypervisor.
```

Comment:
Hypervisor-less virtio requires the presence of a virtio back-end (in the current implementation referred to as a Physical Machine Monitor == PMM) running on a host and providing high level [hypervisor-less] virtio back-end services (and eventually access to its devices) to a remote host. RPMsg is not included in this list of services.

TODO: sync on an appropriate rephrasing.

How is this diagram different from the one in the RemoteProc section?

Reply:
The fastest fix is to remove the last phrase which mentions the "referred to as a hypervisorless-virtio" statement.